schema/deploy/shipping/views.sql

```diff
@@ -557,4 +557,104 @@ create or replace view shipping.observation_with_presence_absence_result_v2 as
 comment on view shipping.observation_with_presence_absence_result_v2 is
   'Joined view of shipping.incidence_model_observation_v3 and shipping.presence_absence_result_v1';
 
+
+create or replace view shipping.sample_with_best_available_site_details_v1 as
+
+    with specimen_manifest_site_data as (
+        select distinct on (manifest_id)
+            cast(value->>'sample_id' as integer) as sample_id,
+            trim(both ' ' from document->>'swab_site') as swab_site,
+            trim(both ' ' from document->>'sample_origin') as sample_origin
+        from
+            receiving.manifest,
+            jsonb_array_elements(processing_log)
+        where value->'sample_id' is not null
+        order by manifest_id desc
```

Comment:
Since this is a `distinct on (manifest_id)`, there should be an ordering on another column as well so the row that's kept is stable. Refer to https://www.postgresql.org/docs/10/sql-select.html#SQL-DISTINCT for more details.

In this case, I think we want the `sample_id` from the _last_ `processing_log` entry, not the first. This can be done by ordering by the log's timestamp field, for example.

Reply:
I addressed this in a new commit.